src/libraries/System.Text.Json/tests/Serialization/PolymorphicTests.cs

```diff
@@ -464,6 +491,22 @@ public static void GenericDictionaryOfInterface_WithValidJson_ThrowsNotSupported
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<MyThingDictionary>(@"{"""":{}}"));
         }
 
+        [Fact]
+        public void AnonymousType()
+        {
+            const string Expected = @"{""x"":1,""y"":true}";
+            var value = new { x = 1, y = true };
+
+            // Strongly-typed.
+            string json = Serializer.Serialize(value);
```

Comment:
Nice test. Could this apply on deserialization too? What about using `value.GetType()` to call the non-generic method?

Reply:
This is not supported as anonymous types don't have parameterless ctors (only a single parameterized ctor). 

This is an area to test in the ctors PR https://github.com/dotnet/runtime/pull/33444, although these types would have to be special cased as they violate the requirement that ctor parameter names should be the camelCase  equivalent of the property names. Instead, they are the same string i.e a property named `Prop` would map to a ctor parameter named `prop`, while `prop` would map to `prop`.